apis/core/bootstrap/shapes/csv-source.ts

```diff
@@ -3,12 +3,65 @@ import { csvw, dash, hydra, rdfs, schema, sh, xsd } from '@tpluscode/rdf-ns-buil
 import { turtle } from '@tpluscode/rdf-string'
 import $rdf from 'rdf-ext'
 
-const shapeId = shape('csv-source/update')
-const csvSourceDialect = $rdf.namedNode(shapeId.value + '#csvSourceDialect')
+const shapeCreateId = shape('csv-source/create')
+const shapeUpdateId = shape('csv-source/update')
+const csvSourceDialect = $rdf.namedNode(shapeUpdateId.value + '#csvSourceDialect')
 
-export const CSVSourceShape = turtle`
-${shapeId} {
-  ${shapeId} a ${sh.NodeShape}, ${hydra.Resource} ;
+export const CSVSourceCreateShape = turtle`
+${shapeCreateId} {
+  ${shapeCreateId} a ${sh.NodeShape}, ${hydra.Resource} ;
+    ${sh.targetClass} ${schema.MediaObject} ;
+    ${rdfs.label} "CSV Source Media Object" ;
+    ${sh.property} [
+      ${sh.name} "File name" ;
+      ${sh.path} ${schema.name} ;
+      ${sh.datatype} ${xsd.string} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+    ] ;
+    ${sh.property} [
+      ${sh.name} "File location URL" ;
+      ${sh.path} ${schema.contentUrl} ;
+      ${sh.nodeKind} ${sh.IRI} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+    ] ;
+    ${sh.property} [
+      ${sh.name} "S3 key" ;
+      ${sh.path} ${schema.identifier} ;
+      ${sh.datatype} ${xsd.string} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+    ] ;
+  .
+
+  ${csvSourceDialect} a ${sh.NodeShape} ;
+    ${rdfs.label} "CSV dialect options" ;
+    ${sh.property} [
+      ${sh.name} "Delimiter" ;
+      ${sh.path} ${csvw.delimiter} ;
+      ${sh.datatype} ${xsd.string} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+      ${sh.minLength} 1 ;
+      ${sh.order} 10 ;
+    ] ;
+    ${sh.property} [
+      ${sh.name} "Quote char" ;
+      ${sh.path} ${csvw.quoteChar} ;
+      ${sh.datatype} ${xsd.string} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+      ${sh.minLength} 1 ;
+      ${sh.order} 20 ;
+    ] ;
+  .
```

Comment:
Copy-pasted? I think this should be removed

Reply:
yup